packages/tailwindcss/src/index.ts

```diff
@@ -33,6 +33,21 @@ function throwOnPlugin(): never {
   throw new Error('No `loadPlugin` function provided to `compile`')
 }
 
+function parseThemeOptions(selector: string) {
+  let isReference = false
+  let isInline = false
+
+  for (let option of segment(selector.slice(6) /* '@theme'.length */, ' ')) {
+    if (option === 'reference') {
+      isReference = true
+    } else if (option === 'inline') {
+      isInline = true
+    }
+  }
+
+  return { isReference, isInline }
+}
+
```

Comment:
@RobinMalfait Decided to extract this function because the work was being done twice, and `themeOptions` didn't feel like a good name for an array that contained:

```
['@theme', 'reference', 'inline']
```

Refactored to a `for` loop to avoid the extra loop that calling `includes` twice would incur, but admittedly sad it's more code.
```

Reply:
Nice, makes sense.

One potential thing we can do is if both options are `true` early return because there is no need to keep looping if everything is found already. But since `@theme` is so specific to Tailwind, there is no need for that optimization (yet).